bump containerd and dependencies to v1.3.0

thaJeztah opened the pull request:
### Update to Containerd v1.3.0

release notes: https://github.com/containerd/containerd/releases/tag/v1.3.0

full diff: https://github.com/containerd/containerd/compare/640860a042b93c26c0a33081ee02230def486f81...v1.3.0

- bump Microsoft/hcsshim 9e921883ac929bbe515b39793ece99ce3a9d7706
    - full diff: https://github.com/Microsoft/hcsshim/compare/672e52e9209d1e53718c1b6a7d68cc9272654ab5...9e921883ac929bbe515b39793ece99ce3a9d7706
- bump hashicorp/golang-lru v0.5.3
    - full diff: https://github.com/hashicorp/golang-lru/compare/v0.5.1...v0.5.3


### Update to Containerd v1.3.0-beta.2

release notes: https://github.com/containerd/containerd/releases/tag/v1.3.0-beta.2

- vendor containerd v1.3.0-beta.2
    - full diff: https://github.com/containerd/containerd/compare/v1.3.0-beta.1...v1.3.0-beta.2
- bump containerd/continuity f2a389ac0a02ce21c09edd7344677a601970f41c
    - full diff: https://github.com/containerd/continuity/compare/aaeac12a7ffcd198ae25440a9dff125c2e2703a7...f2a389ac0a02ce21c09edd7344677a601970f41c
    - no local changes
- bump containerd/fifo bda0ff6ed73c67bfb5e62bc9c697f146b7fd7f13
    - full diff: https://github.com/containerd/fifo/compare/a9fb20d87448d386e6d50b1f2e1fa70dcf0de43c...bda0ff6ed73c67bfb5e62bc9c697f146b7fd7f13


### Update to Containerd v1.3.0-beta.1

release notes: https://github.com/containerd/containerd/releases/tag/v1.3.0-beta.1

- update runc binary v1.0.0-rc8-32-gf4982d86
- bump containerd v1.3.0-beta.1
    - full diff: https://github.com/containerd/containerd/compare/7c1e88399ec0b0b077121d9d5ad97e647b11c870...v1.3.0-beta.1
- bump runc v1.0.0-rc8-32-gf4982d86
    - full diff: https://github.com/opencontainers/runc/compare/425e105d5a03fabd737a126ad93d62a9eeede87f...f4982d86f7fde0b6f953cc62ccc4022c519a10a9
    - opencontainers/runc#2034 Support for logging from children processes
    - opencontainers/runc#2035 specconv: always set "type: bind" in case of MS_BIND
    - opencontainers/runc#2038 `r.destroy` can defer exec in `runner.run` method
    - opencontainers/runc#2042 libcontainer: intelrdt: add missing destroy handler in defer func
    - opencontainers/runc#2047 Move systemd.Manager initialization into a function in that module
    - opencontainers/runc#2057 main: not reopen /dev/stderr
        - closes opencontainers/runc#2056 Runc + podman|cri-o + systemd issue with stderr
        - closes kubernetes/kubernetes#77615 kubelet fails starting CRI-O containers (Ubuntu 18.04 + systemd cgroups driver)
        - closes cri-o/cri-o#2368 Joining worker node not starting flannel or kube-proxy / CRI-O error "open /dev/stderr: no such device or address"
    - opencontainers/runc#2061 libcontainer: fix TestGetContainerState to check configs.NEWCGROUP
    - opencontainers/runc#2065 Fix cgroup hugetlb size prefix for kB
    - opencontainers/runc#2067 libcontainer: change seccomp test for clone syscall
    - opencontainers/runc#2074 Update dependency libseccomp-golang
- bump containerd/go-runc 9007c2405372fe28918845901a3276c0915689a1
    - full diff: https://github.com/containerd/go-runc/compare/7d11b49dc0769f6dbb0d1b19f3d48524d1bad9ad...e029b79d8cda8374981c64eba71f28ec38e5526f
    - containerd/go-runc#52 Fix Method of judging command execution failure
        - fixes "init.pid: no such file or directory: unknown" errors
    - containerd/go-runc#54 avoid setting NOTIFY_SOCKET from calling process
        - addresses moby/moby#39866 User namespace remapping breaks BuildKit-powered builds (19.03.2)

- bump containerd/cgroups c4b9ac5c7601384c965b9646fc515884e091ebb9
    - full diff: github.com/containerd/cgroups https://github.com/containerd/cgroups/compare/4994991857f9b0ae8dc439551e8bebdbb4bf66c1...c4b9ac5c7601384c965b9646fc515884e091ebb9
    - containerd/cgroups#81 Add network stats
        - addresses containerd/cgroups#80 Add network metrics
    - containerd/cgroups#85 Fix cgroup hugetlb size prefix for kB
        - addresses kubernetes/kubernetes#77169 Permission denied on hugetlb due to wrong filename
        - relates to opencontainers/runc#2065 Fix cgroup hugetlb size prefix for kB
    - containerd/cgroups#88 cgroups: fix MoveTo function fail problem
    - containerd/cgroups#92 fixed an issue with invalid soft memory limits
    - containerd/cgroups#93 avoid adding io_serviced and io_service_bytes duplicately
        - fixes containerd/containerd#3412 collected metric container_blkio_io_serviced_recursive_total: was collected before with the same name and label values
- bump containerd/ttrpc
    - full diff: https://github.com/containerd/ttrpc/compare/699c4e40d1e7416e08bf7019c7ce2e9beced4636...1fb3814edf44a76e0ccf503decf726d994919a9a
    - containerd/ttrpc#37 Handle EOF to prevent file descriptor leak
    - containerd/ttrpc#38 Improve connection error handling
    - containerd/ttrpc#40 Support headers
    - containerd/ttrpc#41 Add client and server unary interceptors
    - containerd/ttrpc#42 Refactor close handling for ttrpc clients
    - containerd/ttrpc#43 metadata as KeyValue type

thaJeztah committed:
bump containerd/go-runc e029b79d8cda8374981c64eba71f28ec38e5526f

- github.com/containerd/go-runc https://github.com/containerd/go-runc/compare/7d11b49dc0769f6dbb0d1b19f3d48524d1bad9ad...e029b79d8cda8374981c64eba71f28ec38e5526f
  - containerd/go-runc#52 Fix Method of judging command execution failure
    - fixes "init.pid: no such file or directory: unknown" errors
  - containerd/go-runc#54 avoid setting NOTIFY_SOCKET from calling process

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
Signed-off-by: Derek McGowan <derek@mcgstyle.net>

thaJeztah committed:
bump containerd/cgroups c4b9ac5c7601384c965b9646fc515884e091ebb9

full diff:  github.com/containerd/cgroups https://github.com/containerd/cgroups/compare/4994991857f9b0ae8dc439551e8bebdbb4bf66c1...c4b9ac5c7601384c965b9646fc515884e091ebb9

changes included:

  - containerd/cgroups#81 Add network stats
    - addresses containerd/cgroups#80 Add network metrics
  - containerd/cgroups#85 Fix cgroup hugetlb size prefix for kB
    - addresses kubernetes/kubernetes#77169 Permission denied on hugetlb due to wrong filename
    - relates to opencontainers/runc#2065 Fix cgroup hugetlb size prefix for kB
  - containerd/cgroups#88 cgroups: fix MoveTo function fail problem
  - containerd/cgroups#92 fixed an issue with invalid soft memory limits
  - containerd/cgroups#93 avoid adding io_serviced and io_service_bytes duplicately
    - fixes containerd/containerd#3412 collected metric container_blkio_io_serviced_recursive_total: was collected before with the same name and label values

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Ah! Perhaps I should've built first;

```
21:47:47 vendor/github.com/moby/buildkit/util/leaseutil/manager.go:33:9:warning: cannot use &(local literal) (value of type *local) as leases.Manager value in return statement: missing method AddResource (gosimple)
21:47:47 vendor/github.com/moby/buildkit/util/leaseutil/manager.go:33:9:warning: cannot use &(local literal) (value of type *local) as leases.Manager value in return statement: missing method AddResource (unconvert)
21:47:47 Build step 'Execute shell' marked build as failure
```

thaJeztah commented on vendor.conf:
```diff
@@ -27,7 +27,9 @@ github.com/imdario/mergo                            7c29201646fa3de8506f70121347
-github.com/moby/buildkit                            f5a55a9516d1c6e2ade9bec22b83259caeed3a84 
+
+# WIP WIP WIP using branch used for https://github.com/moby/buildkit/pull/1107
+github.com/moby/buildkit                            containerd-update https://github.com/tonistiigi/buildkit.git
```
temporarily vendoring from https://github.com/moby/buildkit/pull/1107

thaJeztah commented:
Looks like we need `go-cni` now as well (for buildkit)

```
---> Making bundle: binary (in bundles/binary)
Building: bundles/binary-daemon/dockerd-dev
GOOS="" GOARCH="" GOARM=""
vendor/github.com/moby/buildkit/util/network/cni.go:9:2: cannot find package "github.com/containerd/go-cni" in any of:
	/go/src/github.com/docker/docker/vendor/github.com/containerd/go-cni (vendor tree)
	/usr/local/go/src/github.com/containerd/go-cni (from $GOROOT)
	/go/src/github.com/containerd/go-cni (from $GOPATH)
make: *** [binary] Error 1
```

thaJeztah commented on vendor.conf:
```diff
@@ -27,14 +27,18 @@ github.com/imdario/mergo                            7c29201646fa3de8506f70121347
 github.com/mitchellh/hashstructure                  2bca23e0e452137f789efbc8610126fd8b94f73b
 github.com/gofrs/flock                              7f43ea2e6a643ad441fc12d0ecc0d3388b300c53 # v0.7.0
+github.com/containerd/go-cni                        5a4663dad645d9736aa62b366c8a7a07dbe751cc # master
+github.com/containernetworking/cni                  142cde0c766cd6055cc7fdfdcb44579c0c9c35bf # somewhere between v0.6.0 and v0.7.0-alpha0
```
two new dependencies coming from buildkit

@tonistiigi was this intentional (as in; are these needed for buildkit's use in docker, or could we move some code to get rid of the dependencies?)

tonistiigi replied:
you only get them cause you are vendoring buildkit that is older than master(and older than current vendor)

thaJeztah replied:
hmmm; I think I vendored from your PR https://github.com/moby/buildkit/pull/1107 or is that outdated?

thaJeztah commented on daemon/daemon.go:
```diff
@@ -193,12 +193,14 @@ func (daemon *Daemon) NewResolveOptionsFunc() resolver.ResolveOptionsFunc {
 
-		def.PlainHTTP = c.PlainHTTP
+		// TODO ResolverOptions.PlainHTTP is deprecated; ResolverOptions.Hosts should be used
+		def.PlainHTTP = *c.PlainHTTP
```
So changes here are really the hacky approach for now; need to look into the new changes to do it the proper way

thaJeztah committed:
bump containerd/continuity f2a389ac0a02ce21c09edd7344677a601970f41c

full diff: https://github.com/containerd/continuity/compare/aaeac12a7ffcd198ae25440a9dff125c2e2703a7...f2a389ac0a02ce21c09edd7344677a601970f41c

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
bump containerd/fifo bda0ff6ed73c67bfb5e62bc9c697f146b7fd7f13

full diff: https://github.com/containerd/fifo/compare/a9fb20d87448d386e6d50b1f2e1fa70dcf0de43c...bda0ff6ed73c67bfb5e62bc9c697f146b7fd7f13

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Ah, buildkit needs another update, because one change was reverted/changed in 1.3.0-beta.2;

```

[2019-08-26T12:10:01.115Z] # github.com/docker/docker/vendor/github.com/moby/buildkit/util/leaseutil
[2019-08-26T12:10:01.116Z] vendor/github.com/moby/buildkit/util/leaseutil/manager.go:33:16: cannot use &local literal (type *local) as type leases.Manager in return argument:
[2019-08-26T12:10:01.116Z] 	*local does not implement leases.Manager (missing AddResource method)
```

thaJeztah committed:
bump hashicorp/golang-lru v0.5.3

full diff: https://github.com/hashicorp/golang-lru/compare/v0.5.1...v0.5.3

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

dmcgowan committed:
update containerd binary v1.3.0

full diff: https://github.com/containerd/containerd/compare/v1.2.8..v1.3.0

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
Signed-off-by: Derek McGowan <derek@mcgstyle.net>

dmcgowan committed:
bump containerd v1.3.0

full diff: https://github.com/containerd/containerd/compare/7c1e88399ec0b0b077121d9d5ad97e647b11c870...v1.3.0

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
Signed-off-by: Derek McGowan <derek@mcgstyle.net>

dmcgowan commented:
Pending one change in buildkit before this is ready again

thaJeztah commented on hack/dockerfile/install/runc.installer:
I think we can drop this commit, because master is already on this version (this only updates the description)

dmcgowan committed:
bump moby/buildkit f7042823e340d38d1746aa675b83d1aca431cee3

full diff: https://github.com/moby/buildkit/compare/588c73e1e4f0f3d7d3738abaaa7cf8026064b33e...f7042823e340d38d1746aa675b83d1aca431cee3

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

fix daemon for changes in containerd registry configuration

Signed-off-by: Evan Hazlett <ejhazlett@gmail.com>

Update buildernext and daemon for buildkit update

Signed-off-by: Derek McGowan <derek@mcgstyle.net>

dmcgowan commented:
Updated with builtkit change, ping @tonistiigi  for changes related to updated buildkit

thaJeztah commented:
Seeing one failure in the vendor check;

```

[2019-10-03T00:53:28.049Z] The result of vndr differs
[2019-10-03T00:53:28.049Z] 
[2019-10-03T00:53:28.049Z]  D vendor/golang.org/x/sync/singleflight/singleflight.go
[2019-10-03T00:53:28.049Z] 
[2019-10-03T00:53:28.049Z] Please vendor your package with github.com/LK4D4/vndr.
```


Haven't seen this failure on Windows before; https://ci.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-39713/runs/16/nodes/175/log/?start=0


```
[2019-10-03T00:47:34.830Z] === RUN   TestRemoveContainerWithVolume
[2019-10-03T00:47:36.998Z] --- FAIL: TestRemoveContainerWithVolume (1.90s)
[2019-10-03T00:47:36.998Z]     remove_test.go:62: assertion failed: error is not nil: Error response from daemon: process 8908 in container dbfa597498c3335ae3dc5174c5f92336ae5355d4499aa2527ce1bd0a3bbc23c9 encountered an error during hcsshim::Process::StdioLegacy: failure in a Windows system call: Element not found. (0x490)
```

I'll kick the build once more to see if the Windows failure is reproducible

thaJeztah commented:
Triggered Jenkins; I also enabled Windows RS1 on that run, to see if it's working ok on that version (but it might time-out, because RS1 is slow)

thaJeztah commented:
Something weird with the vendor check; just had the same failure on another PR that also bumps containerd and buildkit; https://github.com/moby/moby/pull/40032#issuecomment-537914865

> ok, this is weird; I just tried running `vndr golang.org/x/sync` on this branch, and it did not produce a diff;
> 
> ```
> On branch fix-grpc-withdialer-deprecation-warning
> nothing to commit, working tree clean
> ```
> 
> Also double checked if the base image hasn't been updated on Docker Hub (which doesn't seem to be the case); https://hub.docker.com/_/golang?tab=tags&page=1&name=stretch
> 
> <img width="848" alt="Screenshot 2019-10-03 at 13 59 12" src="https://user-images.githubusercontent.com/1804568/66124991-20f9ee80-e5e6-11e9-83a4-1664214da68b.png">

thaJeztah commented:
OK, looks like it's a problem on master; opened https://github.com/moby/moby/pull/40037 to fix it

andrewhsu commented:
Hmm...I see 5 [errors](https://ci.docker.com/public/job/moby/job/PR-39713/18/execution/node/200/log/) in `TestDockerSuite` from `win-RS5` step:
```
--- FAIL: TestDockerSuite/TestVolumeFromMixedRWOptions (3.80s)
--- FAIL: TestDockerSuite/TestBuildShCmdJSONEntrypoint (3.28s)
--- FAIL: TestDockerSuite/TestBuildWindowsUser (11.35s)
--- FAIL: TestDockerSuite/TestPostContainersCreateWithStringOrSliceCmd (1.73s)
--- FAIL: TestDockerSuite/TestRunEchoStdout (1.86s)
--- FAIL: TestDockerSuite/TestVolumeFromMixedRWOptions (3.80s)
```

dmcgowan commented:
@andrewhsu they all have the same message `encountered an error during hcsshim::Process::StdioLegacy: failure in a Windows system call: Element not found. ` Is this something new or related to updating hcshim?

andrewhsu commented:
I've only seen it here https://github.com/moby/moby/pull/39713#issuecomment-537902850

dmcgowan commented:
Going to try pulling the hcsshim shim commit out for now to see if it still has a breakage while we figure out the source of that error.

thaJeztah reviewed (commented):
all green now; LGTM

cpuguy83 reviewed (approved):
LGTM

tiborvass commented on vendor.conf:
```diff
@@ -117,12 +117,12 @@ github.com/googleapis/gax-go                        317e0006254c44a0ac427cc52a0e
-github.com/containerd/cgroups                       4994991857f9b0ae8dc439551e8bebdbb4bf66c1
+github.com/containerd/containerd                    36cf5b690dcc00ff0f34ff7799209050c3d0c59a # v1.3.0
+github.com/containerd/fifo                          bda0ff6ed73c67bfb5e62bc9c697f146b7fd7f13
+github.com/containerd/continuity                    f2a389ac0a02ce21c09edd7344677a601970f41c
```
not a big deal, but this should be 75bee3e2ccb6402e3a986ab8bd3b17003fc0fdec (see buildkit's go.mod). The only difference is https://github.com/containerd/continuity/compare/f2a389ac0a02ce21c09edd7344677a601970f41c...master#diff-baa2046d08674df161187d3f68336ff6R83

thaJeztah replied:
I think I aligned it to https://github.com/containerd/containerd/blob/v1.3.0/vendor.conf#L7, given that this was listed under "containerd" dependencies. I can open a PR in containerd to update it there.

andrewhsu reviewed (approved):
SGTM

this will get to latest containerd that we need for things to be built on top of
